### PR TITLE
feat: nicer output

### PR DESCRIPTION
**Pull request by karmacoma-eth**
Includes a bunch of things:
- type based counterexample rendering (different rendering for uints, signed ints, address, bool, bytes, string, bytes32...)
- console.log functions leverage the same type based rendering
- includes a bunch of new console.log functions + tests
- updates to latest halmos-cheatcode, implements the new cheatcodes (createInt, createInt256, createBytes4, createString) and add tests for these functions

Plus some internal reorganizing:
- move utils-like functions from sevm.py to utils.py
- move cheatcode and console.log handling logic from sevm.py to cheatcodes.py
- switch to a lookup+handler based mechanism in console and halmos_cheat_code

### CODE REVIEW

File: src/halmos/utils.py [hunk 1]
```diff
@@ -36,13 +213,80 @@ def hexify(x):
...
+    elif type_name == "string":
+        return render_string(val)
+    elif type_name == "bytes":
+        return render_bytes(val)
+    elif type_name == "address":
+        return render_address(val)
+    else:  # bytes32, bytes4, structs, etc.
+        return hexify(val)
+
+
```
Comment (daejunpark) [inline src/halmos/utils.py:289]: i think we need to catch potential exceptions raised by the render functions here.  even if `is_bv_value(val)` is true, the render functions could still raise an exception, when the type of `val` somehow differs from expected, e.g., `.decode("utf-8")` may fail in `render_string()`, or `b.hex()` may fail if `b` is not of type `bytes`.

**Review by daejunpark — COMMENTED** [review]
looks great!  the refactoring is amazing!  thank you so much!

my only comment is about handling potential exceptions in stringify.

File: src/halmos/cheatcodes.py [hunk 2]
```diff
@@ -154,11 +318,262 @@ class hevm_cheat_code:
...
+        0x2D5B6CB9: log_int256,
+        0xB60E72CC: log_string_uint256,
+    }
+
+    @staticmethod
+    def handle(ex, arg: BitVec) -> None:
+        funsig: int = int_of(extract_funsig(arg), "symbolic console function selector")
+
+        if handler := console.handlers.get(funsig):
+            return handler(arg)
```
[the remark this answers is not in the document]
Comment (karmacoma-eth) [reply src/halmos/cheatcodes.py]: you're right, added a test for it in 8b5b151:

```
Running 1 tests for test/Console.t.sol:ConsoleTest
console.handle: UnicodeDecodeError('utf-8', b'\xff', 0, 1, 'invalid start byte') with arg=0x4b5c4277000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000000000002c74686973206973206120737472696e67207468617420776f6e2774206465636f646520746f207574662d383a00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001ff00000000000000000000000000000000000000000000000000000000000000
[PASS] check_log_undecodable_string() (paths: 1/1, time: 0.05s, bounds: [])
Symbolic test result: 1 passed; 0 failed; time: 0.05s
```

File: src/halmos/utils.py (hunk 1 above)
Comment (karmacoma-eth) [reply src/halmos/utils.py:317]: in 5b3c56f

**Review by daejunpark — APPROVED** [review]

**Pull request merged**